olive/passes/onnx/transformer_optimization.py

```diff
@@ -59,6 +59,11 @@ def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigPa
                 ),
             ),
             "use_gpu": PassConfigParam(type_=bool, default_value=is_gpu, description="Flag for GPU inference."),
+            "provider": PassConfigParam(
```

Comment:
![image](https://github.com/microsoft/Olive/assets/13343117/8731b451-dab8-42be-a853-aa105b5fe3f6)
Accelerator spec contains the providers. No need to add it to pass configs.

Reply:
How can I derive the provider parameter from the spec?

Reply:
![image](https://github.com/microsoft/Olive/assets/13343117/204f9bb5-b533-4af6-b729-5fcbc433067a)

`accelerator_spec.execution_provider` should work.

Reply:
I want string 'dml' instead of 'DmlExecutionProvider'. Is it stored there?

Reply:
> I want string 'dml' instead of 'DmlExecutionProvider'. Is it stored there?

Nope, you can just have simple mapping like: 'DmlExecutionProvider' : 'dml'

Reply:
Also quick question, where you are using `dml`?  I just saw the config definition where it is used to optimize model?

Reply:
> Also quick question, where you are using `dml`? I just saw the config definition where it is used to optimize model?

It will be used in `optimize_by_onnxruntime` in ONNX Runtime. 

https://github.com/microsoft/onnxruntime/blob/35ecce45496ee752bc3f85618eba713bb50e6069/onnxruntime/python/tools/transformers/optimizer.py#L62C1-L62C1

Reply:
We should change optimize_by_onnxruntime to directly receive EP list from Olive instead of asking Olive to map EP -> "dml" and then let optimizer.py do "dml" -> EP mapping.